lib/tasks/update_manual_address_entry_selected_prexisting_logs.rake

```diff
@@ -17,12 +22,23 @@ namespace :bulk_update do
         Rails.logger.info "Could not save changes to lettings log #{log.id}"
       end
       status_post_change = log.status
-      unless status_pre_change == status_post_change
-        Rails.logger.info "Status changed from #{status_pre_change} to #{status_post_change} for lettings log #{log.id}"
+      if status_pre_change != status_post_change
+        if log.postcode_full.nil? && log.address_line1 == log.address_line1_input
+          log.postcode_full = log.postcode_full_input
+          log.save!
+        end
+        if log.status == status_pre_change
+          lettings_postcode_fixed_count += 1
+        else
+          Rails.logger.info "Status changed from #{status_pre_change} to #{status_post_change} for lettings log #{log.id}"
+          lettings_status_changed_log_ids << log.id
+        end
       end
     end
 
     puts "#{updated_lettings_logs_count} lettings logs updated."
+    puts "Lettings logs with status changes: [#{lettings_status_changed_log_ids.join(', ')}]"
+    puts "Lettings logs where postcode fix maintained status: #{lettings_postcode_fixed_count}"
```

Comment:
Would it make sense to distinguish between the logs where we updated the postcode and they stayed in progress vs the logs where we didn't even attempt to update the postcode because the address lines didn't match?